### PR TITLE
Allow hostname to be configured

### LINKED ISSUE #79: CNI plugin doesn't support having the hostname configured
It will only use a hostname of `socket.gethostname()` but hosts might be created with different names.

### PR DESCRIPTION
**Pull request by tomdee**
Fixes #79

### CODE REVIEW

File: calico.py [hunk 1]
```diff
@@ -38,6 +38,9 @@
 from calico_cni.policy_drivers import ApplyProfileError, get_policy_driver
 from ipam import IpamPlugin
 
+# Hostname which can be overridden with config
+HOSTNAME = socket.gethostname()
```
Comment (caseydavenport) [inline calico.py]: This value is only used within the `CniPlugin` class.  Why don't we define this within in the class (like the rest of the config options) so we don't need to do this global stuff.

This should reduce the number of lines, and make it easier to manipulate in UTs.

```
self.hostname = config.get("hostname", socket.gethostname())
```

Or something like that.

File: calico.py [hunk 2]
```diff
@@ -686,6 +689,11 @@ def main():
         os.environ[ETCD_ENDPOINTS_ENV] = etcd_endpoints
         _log.debug("Using %s=%s", ETCD_ENDPOINTS_ENV, etcd_endpoints)
 
+    hostname_config = network_config.get("hostname")
```
Comment (caseydavenport) [inline calico.py]: All the other config options use constants for this.

**Comment by tomdee** [comment]
OK, markups applied.

**Comment by caseydavenport** [comment]
LGTM.

**Pull request merged**